beluga_tutorial/src/main.cpp

```diff
@@ -0,0 +1,138 @@
+// https://github.com/Ekumen-OS/beluga/issues/279#issuecomment-1903914387
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <random>
+#include <cmath>
+#include <vector>
+
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/generate.hpp>
+#include <range/v3/view/take_exactly.hpp>
+
+#include <beluga/beluga.hpp>
+
+#define NUM_OF_PARTICLES 200
+
+// TODO: use template to use only one random function.
+bool generateRandomMeasurement(double mean, double sd) {
+    static auto generator = std::mt19937{std::random_device()()};
+    std::normal_distribution<double> distribution(mean, sd);
+
+    // Generate a random number and check if it's greater than 0.5
+    return distribution(generator) > 0.5;
+}
+
+double generateRandomOdom(double mean, double sd) {
+    static auto generator = std::mt19937{std::random_device()()};
+    std::normal_distribution<double> distribution(mean, sd);
+
+    return (distribution(generator));
+}
+
+int main()
+{
+    std::cout << "beluga tutorial - main.cpp start!!" << std::endl;
+
+    // Type of map: 1D map composed by walls or doors
+    struct Landmark {
+        double pose;
+        bool mark; // 0 = wall; 1 = Door
+    };
+
+    // Generate a random map
+    const size_t map_size = 100;
+    std::vector<Landmark> landmark_map;
+    for(size_t i = 0; i < map_size; i++)
+    {
+        Landmark l;
+        l.pose = static_cast<double>(i);
+        l.mark = generateRandomMeasurement(0, 0.5);
+        landmark_map.push_back(l);
+    }
+
+    // Save the map in a csv file
+    std::string filename{"map.csv"};
+    std::fstream fout {filename, fout.trunc | fout.out};
+    fout << "pose,mark" << std::endl;
+    for(const auto& l: landmark_map)
+    {
+        fout << l.pose << ","
+             << l.mark    << std::endl;
+    }
```

Comment:
@Alondruck meta^2: I'm thinking that we could have tutorials print all data to `stdout`. Then we can pipe that output to a Python visualization script, like `./one_dimensional_tutorial | python3 display.py`.